.travis.yml

```diff
@@ -122,10 +122,27 @@ after_success:
         popd;
       fi;
     fi;
-  ## docker: build and push openapi-generator-online to DockerHub
-  - if [ $DOCKER_HUB_USERNAME ]; then echo "$DOCKER_HUB_PASSWORD" | docker login --username=$DOCKER_HUB_USERNAME --password-stdin && docker build -t $DOCKER_GENERATOR_IMAGE_NAME ./modules/openapi-generator-online && if [ ! -z "$TRAVIS_TAG" ]; then docker tag $DOCKER_GENERATOR_IMAGE_NAME:latest $DOCKER_GENERATOR_IMAGE_NAME:$TRAVIS_TAG; fi && if [ ! -z "$TRAVIS_TAG" ] || [ "$TRAVIS_BRANCH" = "master" ]; then docker push $DOCKER_GENERATOR_IMAGE_NAME && echo "Pushed to $DOCKER_GENERATOR_IMAGE_NAME"; fi; fi
-  ## docker: build cli image and push to Docker Hub
-  - if [ $DOCKER_HUB_USERNAME ]; then echo "$DOCKER_HUB_PASSWORD" | docker login --username=$DOCKER_HUB_USERNAME --password-stdin && docker build -t $DOCKER_CODEGEN_CLI_IMAGE_NAME ./modules/openapi-generator-cli && if [ ! -z "$TRAVIS_TAG" ]; then docker tag $DOCKER_CODEGEN_CLI_IMAGE_NAME:latest $DOCKER_CODEGEN_CLI_IMAGE_NAME:$TRAVIS_TAG; fi && if [ ! -z "$TRAVIS_TAG" ] || [ "$TRAVIS_BRANCH" = "master" ]; then docker push $DOCKER_CODEGEN_CLI_IMAGE_NAME && echo "Pushed to $DOCKER_CODEGEN_CLI_IMAGE_NAME"; fi; fi
-
+  ## docker build and push images to DockerHub (openapi-generator-online, openapi-generator-cli)
+  - if [ $DOCKER_HUB_USERNAME ]; then 
+        read -r MVN_VERSION_FOR_DOCKER_TAG < target/ci/version-for-docker-tag.txt
+        echo "Tag for Docker derived from maven version: $MVN_VERSION_FOR_DOCKER_TAG"
+        echo "$DOCKER_HUB_PASSWORD" | docker login --username=$DOCKER_HUB_USERNAME --password-stdin
+        if [ ! -z "$TRAVIS_TAG" ]; then
+          docker build -t $DOCKER_GENERATOR_IMAGE_NAME:$TRAVIS_TAG -t $DOCKER_GENERATOR_IMAGE_NAME:$MVN_VERSION_FOR_DOCKER_TAG ./modules/openapi-generator-online
+          docker build -t $DOCKER_CODEGEN_CLI_IMAGE_NAME:$TRAVIS_TAG -t $DOCKER_GENERATOR_IMAGE_NAME:$MVN_VERSION_FOR_DOCKER_TAG ./modules/openapi-generator-cli
+        elif [ "$TRAVIS_BRANCH" == "master" ]; then
+          docker build -t $DOCKER_GENERATOR_IMAGE_NAME -t $DOCKER_GENERATOR_IMAGE_NAME:$MVN_VERSION_FOR_DOCKER_TAG ./modules/openapi-generator-online
+          docker build -t $DOCKER_CODEGEN_CLI_IMAGE_NAME -t $DOCKER_GENERATOR_IMAGE_NAME:$MVN_VERSION_FOR_DOCKER_TAG ./modules/openapi-generator-cli
+        else
+          docker build -t $DOCKER_GENERATOR_IMAGE_NAME:$MVN_VERSION_FOR_DOCKER_TAG ./modules/openapi-generator-online
+          docker build -t $DOCKER_GENERATOR_IMAGE_NAME:$MVN_VERSION_FOR_DOCKER_TAG ./modules/openapi-generator-cli
+        fi;
+        if [ ! -z "$TRAVIS_TAG" ] || [ "$TRAVIS_BRANCH" == "master" ] || [ "$TRAVIS_BRANCH" == "3.1.x" ] || [ "$TRAVIS_BRANCH" == "4.0.x" ];
+          docker push $DOCKER_GENERATOR_IMAGE_NAME
+          echo "Pushed to $DOCKER_GENERATOR_IMAGE_NAME"; 
+          docker push $DOCKER_CODEGEN_CLI_IMAGE_NAME
+          echo "Pushed to $DOCKER_CODEGEN_CLI_IMAGE_NAME"; 
```

Comment:
If `docker push $DOCKER_GENERATOR_IMAGE_NAME` fails but `docker push $DOCKER_CODEGEN_CLI_IMAGE_NAME` succeeds, is it correct to say that the CI job will still build successfully?

(I think such situation rarely occurs but still want to confirm the behavior in case that happens)

Reply:
Good question.
Did my modification changed this?

---

Was this the reason why there was 2 items in the old version?

```
- if [ $DOCKER_HUB_USERNAME ]; ... => DOCKER_GENERATOR_IMAGE_NAME
- if [ $DOCKER_HUB_USERNAME ];  ... => DOCKER_CODEGEN_CLI_IMAGE_NAME
```

---

My idea to bring both together was to reduce copy-paste code, but I can split it back to 2 items.

Reply:
> Was this the reason why there was 2 items in the old version?

I wasn't the one who added it but that could be the reason.

> My idea to bring both together was to reduce copy-paste code, but I can split it back to 2 items.

I definitely like your idea and formatting to make the 1-liner bash command more readable.

What about checking the exit status of line 141 before proceeding further?

Reply:
> What about checking the exit status of line 141 before proceeding further?

Good Idea, am not not really an expert in this domain.